bundle/lib/source/DobbySpecConfig.cpp

```diff
@@ -2580,3 +2507,108 @@ bool DobbySpecConfig::processCapabilities(const Json::Value& value,
 
     return true;
 }
+
+// -----------------------------------------------------------------------------
+/**
+ *  @brief Inserts rdkPlugin json into existing json
+ *
+ *  Instead of blindly overwriting the plugin data, we only overwrite sections
+ *  of the plugin data that have been set in the rdkPlugin field of the spec.
+ *
+ *  This allows us to set smaller portions of the plugin data and merge them
+ *  with the data set by the processor methods.
+ *
+ *  @param[in]  value       The rdkPlugins field from the json spec
+ *  @param[in]  dictionary  Pointer to the OCI dictionary to populate
+ *
+ *  @return true if correctly processed the value, otherwise false.
+ */
+void DobbySpecConfig::insertIntoRdkPluginJson(const std::string& pluginName,
+                                              const Json::Value& pluginData)
+{
+    Json::Value& existingData = mRdkPluginsJson[pluginName]["data"];
+
+    // iterate through all data members in the RDK plugin's data field
+    for (auto dataMember : pluginData.getMemberNames())
```

Comment:
Same with all these for loops - any reason not to use `const auto&` here since you're not modifying the contents of dataMember and I see no real reason to make copies?